lib/bedrock/data_plane/commit_proxy/finalization.ex

```diff
@@ -454,34 +457,20 @@ defmodule Bedrock.DataPlane.CommitProxy.Finalization do
 
   @spec build_transactions_for_logs(FinalizationPlan.t(), %{Log.id() => [Bedrock.range_tag()]}) ::
           {:ok, %{Log.id() => Transaction.encoded()}} | {:error, term()}
-  defp build_transactions_for_logs(plan, logs_by_id) do
-    aborted_set = MapSet.new(plan.aborted_indices)
+  defp build_transactions_for_logs(%{} = plan, logs_by_id) do
+    initial_mutations_by_log = Map.new(logs_by_id, fn {key, _} -> {key, []} end)
 
-    initial_mutations_by_log =
-      logs_by_id
-      |> Map.keys()
-      |> Map.new(&{&1, []})
-
-    case plan.transactions
-         |> Enum.with_index()
-         |> Enum.reduce_while(
-           {:ok, initial_mutations_by_log},
-           fn transaction_with_idx, {:ok, acc} ->
-             process_transaction_for_logs(
-               transaction_with_idx,
-               aborted_set,
-               plan.storage_teams,
-               logs_by_id,
-               acc
-             )
-           end
-         ) do
+    plan.indexed_transactions
+    |> Enum.reduce_while({:ok, initial_mutations_by_log}, fn transaction_with_idx, {:ok, acc} ->
+      process_transaction_for_logs(transaction_with_idx, plan.aborted_indices_set, plan.storage_teams, logs_by_id, acc)
+    end)
+    |> case do
       {:ok, mutations_by_log} ->
         result =
           Map.new(mutations_by_log, fn {log_id, mutations_list} ->
             encoded =
               Transaction.encode(%{
-                mutations: Enum.reverse(mutations_list),
+                mutations: Enum.reverse(mutations_list), # Why reverse here?
```

Comment:
because `clear` before `set` is not the same as `set` before `clear`

Reply:
But aren't these mutations conflict-free at this point?